### ISSUE TITLE
Rationale for `copysign` in normalized Quat constructor?

### ISSUE THREAD

**Issue by rdeits** [FugroRoames/Rotations.jl#44]
@tkoolen and I noticed that the `copysign` in the normalized `Quat` constructor was causing issues for integration (since it makes the individual components of the quaternion non-differentiable). Removing the `copysign` entirely fixed my issue, but I'm wondering if there's some other reason for why it should be there? If not, then removing it would make my life easier.

**Comment by tkoolen** [comment]
Removing the `copysign` currently results in test failures in:
* https://github.com/FugroRoames/Rotations.jl/blob/525bbd7692f6a50ccf74603dd3040a0c4c79209a/test/rotation_tests.jl#L186
* https://github.com/FugroRoames/Rotations.jl/blob/525bbd7692f6a50ccf74603dd3040a0c4c79209a/test/derivative_tests.jl#L29
* https://github.com/FugroRoames/Rotations.jl/blob/525bbd7692f6a50ccf74603dd3040a0c4c79209a/test/derivative_tests.jl#L62
* https://github.com/FugroRoames/Rotations.jl/blob/525bbd7692f6a50ccf74603dd3040a0c4c79209a/test/derivative_tests.jl#L157

We can take a look at these tomorrow.

**Comment by rdeits** [comment]
The following changes seem to fix all the test errors, but I'm not sure if this is the right thing to do: https://github.com/FugroRoames/Rotations.jl/compare/master...rdeits:no-copysign?expand=1

**Comment by tkoolen** [comment]
That looks very reasonable.

**Comment by rdeits** [comment]
We should probably come up with some test cases that this change would actually *fix* though ;-)

**Comment by tkoolen** [comment]
@c42f, @andyferris, any thoughts?

**Comment by andyferris** [comment]
I'm trying to remember...

The reason may have been that we need to ensure that we are modelling rotations not quaternions, so they need to obey the `AbstractMatrix` definitions for norms, distances and `==` (etc). Perhaps this was a way of simplifying the fact that quaternions are a double cover of rotations?

@awbsmith do you remember anything about this? 

I'm happy for any improvements here. :)

**Comment by awbsmith** [comment]
Yeah it was about the double cover of quaternions (q = -q) and keeping the rotation angle in the range [-180, 180].   From memory it went in because the mapping between Quat and SpQuat has a singularity at q = [-1.0,0,0,0] (its "nicer" the further you are from that singularity).

Apart from that I'd check ```q == -q``` and  that allowing q.w < 0 doesn't break anything related to interpolation or rotation between coordinate frames, i.e. the minimum rotation between coordinate frames with orientations q and -q.

**Comment by ryanelandt** [comment]
I too am running into issues integrating with quaternions. Modifying the Quat to SPQuat function (https://github.com/FugroRoames/Rotations.jl/compare/master...ryanelandt:master) can avoid the singularity when w is negative. I think that q == -q is always true because the == function compares the elements of the corresponding rotations matrices.
